### PR TITLE
Add a script for running e2e tests for kubetest2 gke deployer

### PR DESCRIPTION
**Pull request by chizhg**
Also BTW fix one issue for kubetest2 gke retry when manually testing the script.

Fixes https://github.com/kubernetes-sigs/kubetest2/issues/152

### CODE REVIEW

File: kubetest2-gke/ci-tests/buildupdown.sh [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      *)
+        echo "Invalid argument"
+        exit 1
+        ;;
+      esac
+    shift
+  done
+
+  NUM_CLUSTERS=0
+  case "${CLUSTER_TOPOLOGY}" in
```
Comment (amwat) [inline kubetest2-gke/ci-tests/buildupdown.sh:54]: nit: I assume this stands for single cluster versus multi cluster? let's make that more verbose?

File: kubetest2-gke/ci-tests/buildupdown.sh [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    --strategy "${BUILD_STRATEGY}" \
+    --stage gs://kubernetes-jenkins/ci \
+    --num-clusters "${NUM_CLUSTERS}" \
+    --num-nodes 1 \
+    --zone us-central1-c,us-west1-a,us-east1-b \
+    --build \
+    --up \
+    --down
+}
+
```
Comment (amwat) [inline kubetest2-gke/ci-tests/buildupdown.sh:79]: let's also add a exec tester and a lightweight test like `kubectl cluster-info dump`

**Review by amwat — COMMENTED** [review]
/approve

File: kubetest2-gke/ci-tests/buildupdown.sh (hunk 2 above)
Comment (chizhg) [reply kubetest2-gke/ci-tests/buildupdown.sh:80]: Done.

File: kubetest2-gke/ci-tests/buildupdown.sh (hunk 1 above)
Comment (chizhg) [reply kubetest2-gke/ci-tests/buildupdown.sh:54]: Done.

File: kubetest2-gke/ci-tests/buildupdown.sh [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  kubetest2 gke \
+    -v 2 \
+    --repo-root "$REPO_ROOT" \
+    --strategy "${BUILD_STRATEGY}" \
+    --stage gs://kubernetes-jenkins/ci \
+    --num-clusters "${NUM_CLUSTERS}" \
+    --num-nodes 1 \
+    --zone us-central1-c,us-west1-a,us-east1-b \
+    --build \
```
Comment (amwat) [inline kubetest2-gke/ci-tests/buildupdown.sh:75]: more relevant here, let's drop build altogether and just have up down. we won't have the right sources in upstream anyway.

File: kubetest2-gke/ci-tests/buildupdown.sh [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+set -o nounset
+set -o pipefail
+set -o xtrace
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+cd "${REPO_ROOT}" &> /dev/null || exit 1
+
+make install
+make install-deployer-gke
+
```
Comment (amwat) [inline kubetest2-gke/ci-tests/buildupdown.sh:27]: need to install the exec tester here

File: kubetest2-gke/ci-tests/buildupdown.sh (hunk 3 above)
Comment (chizhg) [reply kubetest2-gke/ci-tests/buildupdown.sh]: Got it. Removed.

File: kubetest2-gke/ci-tests/buildupdown.sh (hunk 4 above)
Comment (chizhg) [reply kubetest2-gke/ci-tests/buildupdown.sh]: Added.

**Comment by amwat** [comment]
let's merge the prow job first so that this get exercised

**Comment by chizhg** [comment]
This PR should be ready to be merged.

**Pull request merged**